02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -17,43 +17,41 @@ It is well-known among the PureScript community that its documentation is lackin
 Outline
 - Explain what "good" documentation is and define a criteria
 - Evaluate PureScript's documentation using that criteria
-- Explain why PureScript's documentation is lacking and what is being done to improve it
+- Explore where PureScript's documentation is lacking and what can be done to improve it
 - Address various audience's possible concerns or questions centered on these themes:
     - New learners - How should I learn PureScript?
     - PureScript documentation writers - How should I write good maintainable documentation for others?
-    - Core Contributors -
+    - Core Contributors - 
 
 ## What is "Good" Documentation Anyway?
 
-Did someone ever teach you how to write "good" documentation? No, you likely just wrote what came to mind and hoped it was good enough.
-
-So, it helps to understand what is "good" documentation and why.
+Did someone ever teach you how to write "good" documentation? Probably not - you likely just wrote what came to mind and hoped it was good enough. So, it's useful to learn a definition of good documentation and understand why it's defined that way.
 
 Essentially, there are 3 factors that affect whether documentation is "good" or not.
-1. The type of documentation
-2. The intended audience
-3. How up-to-date the documentation is
+1. The intended audience
+2. The type of documentation: the question being answered, targeting a specific subsection of the audience
+3. How accurately it reflects the desired version of the thing being documented
```

Comment:
Hmm... I think that having the intended audience come before the type makes more sense. Good idea!

> How accurately it reflects the desired version of the thing being documented

I think this rendering explains the idea I was trying to convey better. However, could this be reworded or rendered differently somehow. I feel like it's still a bit too complicated. Maybe "How accurately it documents the desired version of the code/project" ? "thing" isn't as concrete as 'code'/'project', especially in this context

Reply:
I like the term "reflects" because I believe it properly relates documentation to the thing its documenting. A thing and its reflection should be "the same" in some sense. How a thing behaves and how its documentation reflects that behavior should be "the same", but the documentation is expected to add, for example, information and context to make the behavior easy to understand and remember.

I think both are fine here, however, and I like how you cleaned it up.

- "How accurately it reflects the desired version of the code/project"
- "How accurately it documents the desired version of the code/project"

Reply:
Mm... good point on the 'reflection' concept. 

However, "it" in this context is not that clear. In the first two points above this one, we never mention "it" If we were more specific with what 'it' means it would look like this:
"How accurately the documentation reflects the desired version of the code/project" ?

However, once it's rendered like that, it seems odd to describe 'documentation' as 'reflecting' something rather than 'explaining' it. Does it still work? Yes, but I don't think it's the best rendering still.

One way we could address this is to use 'it' in every point, so that it now reads (full context):

<hr>

Essentially, there are 3 factors that affect whether documentation is "good" or not.
1. Its intended audience
2. Its type: the questions being answered, targeting a specific subsection of the audience
3. How accurately it reflects the desired version of the code/project

Reply:
Very nice. I've updated to use "it".